Resources/Prototypes/Entities/Objects/Fun/crayons.yml

```diff
@@ -152,15 +152,22 @@
     - id: CrayonBlue
     - id: CrayonPurple
     - id: CrayonBlack
+  - type: ItemCounter
   - type: Appearance
     visuals:
-    - type: StackVisualizer
-      composite: true
-      stackLayers:
-      - red_box
-      - orange_box
-      - yellow_box
-      - green_box
-      - blue_box
-      - purple_box
-      - black_box
+      - type: MappedItemVisualizer
+        mapLayers:
+          - layer: red_box
+            id: CrayonRed
+          - layer: orange_box
+            id: CrayonOrange
+          - layer: yellow_box
+            id: CrayonYellow
+          - layer: green_box
+            id: CrayonGreen
+          - layer: blue_box
+            id: CrayonBlue
+          - layer: purple_box
+            id: CrayonPurple
+          - layer: black_box
+            id: CrayonBlack
```

Comment:
Gonna need to rethink this because it doesn't work with prototype inheritance at all.